gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -730,6 +726,11 @@ public SeekableByteChannel position(long newPosition) throws IOException {
     cancelCurrentRequest();
     invalidateBufferedContent();
 
+    if (footerBuffer == null && newPosition > 0 && newPosition >= footerStartOffsetInBytes) {
```

Comment:
Should add the prefetch cost to `read` instead of `position` ? That would enable us to cache partial overlap with footer data as well ?

Reply:
moved the prefetch to read under different condition. please review. 
I checked the parquet file format. It contains the footer length right at the end. So we may not be able to cache partial overlap in case footer is larger than our `rangerequestsize`. The first seek will happen right at the end of the footer.